packages/@react-stately/overlays/src/useOverlayTriggerState.ts

```diff
@@ -40,7 +40,7 @@ export function useOverlayTriggerState(props: OverlayTriggerProps): OverlayTrigg
       setOpen(false);
     },
     toggle() {
-      setOpen(!isOpen);
+      setOpen(wasOpen => !wasOpen);
```

Comment:
This happened too fast in the test and it was using the old isOpen value, so this makes that impossible, it'll always use the previous value of isOpen